src/Input/Silk.NET.Input.Desktop/GlfwCursor.cs

```diff
@@ -11,23 +13,147 @@ namespace Silk.NET.Input.Desktop
     /// <summary>
     /// A GLFW-based mouse cursor.
     /// </summary>
-    public class GlfwCursor : ICursor
+    internal class GlfwCursor : ICursor, IDisposable
     {
+        private static readonly Dictionary<StandardCursor, CursorShape> CursorShapes = new Dictionary<StandardCursor, CursorShape>
+        {
+            { StandardCursor.Arrow, CursorShape.Arrow },
+            { StandardCursor.IBeam, CursorShape.IBeam },
+            { StandardCursor.Crosshair, CursorShape.Crosshair },
+            { StandardCursor.Hand, CursorShape.Hand },
+            { StandardCursor.HResize, CursorShape.HResize },
+            { StandardCursor.VResize, CursorShape.VResize },
+        };
+        private const int BytesPerCursorPixel = 4;
+
+        private readonly unsafe WindowHandle* _handle;
+        private unsafe Cursor* _cursor = null;
+        private CursorType _cursorType = CursorType.Standard;
+        private StandardCursor _standardCursor = StandardCursor.Default;
+        private int _hotspotX = 0;
+        private int _hotspotY = 0;
+        private int _width = 0;
+        private int _height = 0;
+        private byte[] _pixels = null;
+
+        internal unsafe GlfwCursor(WindowHandle* handle)
+        {
+            _handle = handle;
+        }
+
         /// <inheritdoc />
-        public int HotspotX { get; set; }
-        
+        public unsafe CursorType Type
+        {
+            get => _cursorType;
```

Comment:
It doesn't look like these fields are ever updated by GLFW itself. For example, what if the hotspot changed on GLFW's accord? Silk would never respond to that change. Please adjust to update where appropriate. This could be one update every frame or you could subscribe to an event within the GlfwMouse Subscribe method.

Reply:
You mean when a standard cursor is set? Or do you mean someone invokes glfw functions outside of our window class? Note that a cursor is always bound to a specific window. To get the windows handle and change the cursor on its own, someone must do really evil things.

I see your point that the hotspot and dimensions also should reflect the current state for standard cursors but no other case comes to mine. What did you have in mind when you said "if the hotspot changed on GLFW's accord"?

Reply:
The hotspot is likely to change. As a result, we should be keeping HotspotX and HotspotY up-to-date. If you can think of any other properties we should also keep up-to-date, we should update them too. Either subscribe to an event using GlfwEvents or hook into the update loop as seen in GlfwJoystick

Reply:
The hotspot should not change. It is a constant relative offset which stays the same as long as the cursor shape is the same. Think of the hand cursor. The pointing finger is not in the upper-left corner (in contrast to the arrow cursor). So the hand cursor has a hotspot of like { x: 4, y: 0}. These values will not change unless the shape is changed. Cursor movement has no effect on the hotspot.

Reply:
Hmmm well are there any other properties you think that might be of concern?

Reply:
Not really. They will only change on user action in code. The only thing is that hotspot and width will not reflect the correct values if a standard cursor is active. I will think about how to handle this nicely.

Reply:
Is that a goal for this PR or a future PR?

Reply:
I guess it is not a big deal to keep it like it is. I will create a new PR later if I have a good solution. I think this PR can be merged. If you want we can rename the two mentioned methods but I don't know if I will have time for it in the next days.

Reply:
Yeah let's rename the overloads and then I'll merge this in